app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt

```diff
@@ -229,10 +233,23 @@ fun ActivityDetailsScreen(
                   modifier = Modifier.testTag("location")) {
                     Icon(Icons.Default.LocationOn, contentDescription = "Location")
                     Spacer(modifier = Modifier.width(SMALL_PADDING.dp))
-                    Text(
-                        text = location?.name ?: "No location",
-                        modifier = Modifier.testTag("locationText"))
+                    Column {
+                      Text(
+                          text = location?.name ?: "No location",
+                          modifier = Modifier.testTag("locationText"))
+                      if (distance != null) {
+                        val distanceString =
+                            "Distance : " +
+                                if (distance < 1) {
+                                  "${round(distance * 1000)}m"
```

Comment:
Great detail ! Indeed, for less than 1km distance, it is better to show the distance in meters, with a better precision. Great ! However, you did not implement tests that check if this works correctly...